src/Analyzers/MSTest.Analyzers/UseAttributeOnTestMethodAnalyzer.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+
+using Analyzer.Utilities.Extensions;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using MSTest.Analyzers.Helpers;
+
+namespace MSTest.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class UseAttributeOnTestMethodAnalyzer : DiagnosticAnalyzer
+{
+    internal static readonly DiagnosticDescriptor OwnerRule = DiagnosticDescriptorHelper.Create(
+        DiagnosticIds.AttributeOnTestMethodRuleId,
+        title: new LocalizableResourceString(
+            nameof(Resources.UseAttributeOnTestMethodAnalyzerTitle), Resources.ResourceManager, typeof(Resources), GetShortAttributeName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingOwnerAttribute)),
+        messageFormat: new LocalizableResourceString(
+            nameof(Resources.UseAttributeOnTestMethodAnalyzerMessageFormat), Resources.ResourceManager, typeof(Resources), GetShortAttributeName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingOwnerAttribute)),
```

Comment:
It's quite some extra complexity to compute the same split over and over again. I maintain my initial though that the sharing part is adding more complexity that it brings simplicity.

Reply:
I agree that this is pretty complex for a rule that most users will never see fail, but will always run in the background. Please either ensure that it does no allocations (e.g. the .Split to figure out a known string), or simplify this rule. I would be totally happy with reporting a single line that lists all the attributes that are wrong. Or say "Attribute can only be used on [TestMethod]." and link to help that lists all the attributes.

I would lean on Amaury's expertise here to optimize this :)

Reply:
Removed the split and instead hardcoded the attribute short name string, as per my offline discussion with Amaury.